tools/java/org.hl7.fhir.tools.core/src/org/hl7/fhir/tools/implementations/ember/EmberGenerator.java

```diff
@@ -0,0 +1,278 @@
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+ */
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.*;
+
+import org.hl7.fhir.definitions.model.Definitions;
+import org.hl7.fhir.definitions.model.ResourceDefn;
+import org.hl7.fhir.definitions.model.TypeDefn;
+import org.hl7.fhir.tools.implementations.BaseGenerator;
+import org.hl7.fhir.tools.publisher.FolderManager;
+import org.hl7.fhir.tools.publisher.PlatformGenerator;
+import org.hl7.fhir.utilities.Logger;
+import org.hl7.fhir.utilities.Utilities;
+import org.hl7.fhir.utilities.ZipGenerator;
+import org.stringtemplate.v4.ST;
+import org.stringtemplate.v4.STGroup;
+import org.stringtemplate.v4.STRawGroupDir;
+
+public class EmberGenerator extends BaseGenerator implements PlatformGenerator {
+
+    @Override
+    public String getName() {
+        return "ember";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Ember";
+    }
+
+    @Override
+    public String getDescription(String version, String svnRevision) {
+        return "Generates ember models for FHIR resources";
+    }
+
+    @Override
+    public String getVersion() {
+        return "0.1";
+    }
+
+    @Override
+    public boolean isECoreGenerator() {
+        return false;
+    }
+
+    @Override
+    public void generate(Definitions definitions, String destDir, String implDir, String version, Date genDate, Logger logger, String svnRevision)
+            throws Exception {
+        final String basedDir = Utilities.path(implDir, "base");
+
+        Map<String, String> dirs = new HashMap<String, String>() {{
+            put("modelDir", Utilities.path(basedDir, "app", "models"));
+            put("serializerDir", Utilities.path(basedDir, "app", "serializers"));
+        }};
+
+        createDirStructure(dirs);
+
+        STGroup templateGroup = new STRawGroupDir(Utilities.path(basedDir, "templates"));
+
+        Map<String, TypeDefn> typeDefs = definitions.getTypes();
+        for (String name : typeDefs.keySet()) {
+            generateEmberModel(name, dirs.get("modelDir"), templateGroup, definitions);
+        }
+
+        Map<String, TypeDefn> infDefs = definitions.getInfrastructure();
+        for (String name : infDefs.keySet()) {
+            generateEmberModel(name, dirs.get("modelDir"), templateGroup, definitions);
+        }
+
+        Map<String, TypeDefn> structs = definitions.getStructures();
+        for (String name : structs.keySet()) {
+            generateEmberModel(name, dirs.get("modelDir"), templateGroup, definitions);
+        }
+
+        Map<String, ResourceDefn> namesAndDefinitions = definitions.getResources();
+        // generateGoRouting(namesAndDefinitions, dirs.get("serverDir"));
```

Comment:
I'd suggest deleting all of the leftover go-related stuff, including this commented code (and any other related commented code).